cmd/promtool/backfill.go

```diff
@@ -51,6 +52,16 @@ func (p *OpenMetricsParser) Next() (textparse.Entry, error) {
 		if et, err := p.Parser.Next(); err != io.EOF {
 			return et, err
 		}
+		if string(line) == "# EOF\n" {
```

Comment:
The \n is optional.

Reply:
We add it here: https://github.com/prometheus/prometheus/pull/8232/files#diff-1f356c3850c952097aff258debcd0f10d6a313b616a17510de7d6c0448b4d853R49

Reply:
That's not obvious, the unittests need to cover this.

In general I think it's unwise to hack the parsing like this - it bypasses not just this check, but future checks that will work across lines.